tests/functional/src/main/java/org/mule/functional/classloading/isolation/classloader/MuleClassLoaderRunnerFactory.java

```diff
@@ -0,0 +1,265 @@
+
+        List<ArtifactClassLoader> extensionPluginsArtifactClassLoaders = new ArrayList<>();
+        if (!artifactUrlClassification.getPluginClassificationURLs().isEmpty())
+        {
+            classLoader = createPluginClassLoaders(classLoader, childClassLoaderLookupPolicy, artifactUrlClassification, extensionPluginsArtifactClassLoaders);
+        }
+
+        ArtifactClassLoader appClassLoader = createApplicationArtifactClassLoader(classLoader, childClassLoaderLookupPolicy, artifactUrlClassification);
+
+        return new ClassLoaderTestRunner(containerClassLoader, extensionPluginsArtifactClassLoaders, appClassLoader);
+    }
+
+    /**
+     * Creates an {@link ArtifactClassLoader} for the container. As difference from a mule container this one has to be aware that the parent class loader
+     * has all the URLs loaded in launcher app class loader so it has to create a particular look policy to resolve classes as CHILD_FIRST.
+     * In order to do that a {@link FilteringArtifactClassLoader} is created to similates the launcher class loader to resolve only classes from the URLs defined
+     * to be in the container.
+     *
+     * @param testContainerClassLoaderFactory {@link TestContainerClassLoaderFactory} that has the logic to create a container class loader
+     * @param artifactUrlClassification the url classifications to get plugins urls
+     * @return an {@link ArtifactClassLoader} for the container.
+     */
+    private ArtifactClassLoader createContainerArtifactClassLoader(TestContainerClassLoaderFactory testContainerClassLoaderFactory, ArtifactUrlClassification artifactUrlClassification)
+    {
+        logClassLoaderUrls("CONTAINER", artifactUrlClassification.getContainerURLs());
+        MuleArtifactClassLoader launcherArtifact = new MuleArtifactClassLoader("launcher", new URL[0],
+                                                                               MuleClassLoaderRunnerFactory.class.getClassLoader(), new MuleClassLoaderLookupPolicy(Collections.emptyMap(), Collections.<String>emptySet()));
+        ClassLoaderFilter filteredClassLoaderLauncher = new ContainerClassLoaderFilterFactory().create(testContainerClassLoaderFactory.getBootPackages(), Collections.<MuleModule>emptyList());
+
+        return testContainerClassLoaderFactory.createContainerClassLoader(new FilteringArtifactClassLoader(launcherArtifact, filteredClassLoaderLauncher));
+    }
+
+    /**
+     * For each plugin defined in the classification it will create an {@link ArtifactClassLoader} with the name defined in classification.
+     * For extension plugins it will also create the filter based on the extension manifest file. For a plain plugin it will collect the exported
+     * packages and resources for creating the filter from its mule-module.properties file.
+     * <pr/>
+     * It also creates a sharedLibs plugin without any library so far, in order to be a similar representation as mule's container has.
```

Comment:
Basically when the mule-file.xsd resource is loaded, the extension plugin is used to resolve the resource but later the MuleDocumentLoader needs to load mule-extension.xsd but it cannot found it. If the sahredLib is added this one is used to get the resource from its parent (mule->launcher) and loads the xsd. I confirmed that this works in the same way on standalone mode, that was the reason why I added this "empty sharedlibs CL"